### PR TITLE
limit number of ifname to 15

### PR DESCRIPTION
**Pull request by PolynomialDivision**

### CODE REVIEW

**Comment by spolack** [comment]
das sieht mir weit entfernt von fertig und kaputt aus. Bitte solche Changes gut testen.

File: roles/cfg_openwrt/templates/common/config/wireless.j2 [hunk 1]
```diff
@@ -93,7 +93,7 @@ config wifi-iface '{{ wd_id }}_if{{ loop.index0 }}'
       {% if 'mcast_rate' in iface %}
 	option mcast_rate '{{ iface['mcast_rate'] }}'
       {% endif %}
-	option network '{{ mesh_net['name'] }}'
+	option network '{{ mesh_net['name'] }}[:15]'
```
[the remark this answers is not in the document]
Comment (PolynomialDivision) [reply roles/cfg_openwrt/templates/common/config/wireless.j2:96]: jo ist falsch xD

**Comment by pmelange** [comment]
Is it possible for the truncation to cause problems?  For example, would two names such as ```abcdefghijklmnopqrstuvwxyz1``` and ```abcdefghijklmnopqrstuvwxyz2``` both be truckated to ```abcdefghijklmno```?

**Comment by spolack** [comment]
Indeed it is! but we could implement an assertion, assuring we wont have duplicate resulting interface names.

tbh: i'd prefer to document the current behavior and leave it as it is. keep it simple is the way to go IMHO

**Comment by pmelange** [comment]
How about a check to see if a name is over 15 chars, and if so, error.  Maybe with an error message like "ERROR: 0x58F5a2 Don't be a fool"

**Comment by robertfoss** [comment]
@pmelange  I like this solution. Fail hard with a clear error.

**Comment by robertfoss** [comment]
I tested out this solution, and it seems to work fine. It'll throw an error when the name is too long, and not cause any issues for shorter names.


```
        {{ ('Network name ok' if mesh_net['name'] | length < 15) | mandatory('Network name is longer than 15 characters') }}
	option network '{{ mesh_net['name'] }}'
```

I'm not able to add changes to this PR, but this is the way I'd go about it.

**Comment by PolynomialDivision** [comment]
Just make a new PR? :)

**Comment by robertfoss** [comment]
Created #137 which implements the above solution.

**Pull request closed without merging**